dm/syncer/syncer_test.go

```diff
@@ -1703,3 +1687,72 @@ func (s *testSyncerSuite) TestExecuteSQLSWithIgnore(c *C) {
 
 	c.Assert(mock.ExpectationsWereMet(), IsNil)
 }
+
+func genDefaultSubTaskConfig4Test() *config.SubTaskConfig {
```

Comment:
this is not an unit test covers https://github.com/pingcap/tiflow/pull/4035#discussion_r793311221

What I want is only to test `waitBeforeRunExit`, not whole Syncer. I think you can construct a simple Syncer, call `waitBeforeRunExit(ctx1)`, cancel ctx1, wait `maxPauseOrStopWaitTime`, check if `runCtx` is canceled